aspnetcore/fundamentals/dependency-injection.md

```diff
@@ -220,6 +220,38 @@ In the following example, the call to `AddSingleton` registers `MyDependency` as
 services.AddSingleton<IMyDependency, MyDependency>();
 // The following line has no effect:
 services.TryAddSingleton<IMyDependency, DifferentDependency>();
+
+public class MyService
+{
+    public MyService(IMyDependency myDependency, 
+        IEnumberable<IMyDependency> myDependencies)
+    {
+        Trace.Assert(myDependency is MyDependency);
+        Trace.Assert(myDependencies.Single() is MyDependency);
+    }
+}
+```
+
+`DifferentDependency` replaces `MyDependency` if the call to `TryAddSingleton` is changed to a second call to `AddSingleton`, unless multiple `IMyDependency` instances are resolved via `IEnumerable<IMyDependency>`. In that case, the second call to `AddSingleton` adds `DifferentDependency` after `MyDependency` in the `IEnumerable<IMyDependency>`.
+
+In the following example, the second call to `AddSingleton` overrides the previous one or adds to the previous one when multiple services are resolved via `IEnumerable<IMyDependency>`:
```

Comment:
```suggestion
In the following example, the second call to `AddSingleton` overrides the previous one when resolved as `IMyDependency` and adds to the previous one when multiple services are resolved via `IEnumerable<IMyDependency>`:
```

Reply:
What if I replace all of the prose above with the following? It's quite the long sentence but is probably still easier to grok than what I had before.

```suggestion
In the following example, the call to `TryAddSingleton` is changed to a second call to `AddSingleton` which overrides the previous one when resolved as `IMyDependency` and adds to the previous one when multiple services are resolved via `IEnumerable<IMyDependency>`:
```